opm/parser/eclipse/Utility/String.hpp

```diff
@@ -19,6 +20,18 @@ typename std::decay< T >::type uppercase( T&& x ) {
     return uppercase( t, t );
 }
 
+
+template<typename T>
+std::string trim_copy(const T& s) {
+    const auto c_str = s.c_str();
+    const auto end_pos = std::strchr(c_str, ' ');
+    if (end_pos == nullptr)
+        return std::string{ c_str };
+    else {
+        const std::string::size_type length = end_pos - c_str;
+        return std::string{ c_str, length };
+    }
```

Comment:
If `T` here is supposed to be something that has a `c_str()`, then you could really implement this as
```C++
template <typename StringType>
std::string trim_copy(const StringType& input)
{
    auto ret = std::string(input.c_str());

    const auto end = ret.find_last_not_of(" \t\n\r\f\v");
    if (end == std::string::npos)
        return ret;
    else if (end == 0)
        return "";

    return ret.substr(0, end);
}
```